Wyam/config.wyam

```diff
@@ -1,16 +1,16 @@
-#n Wyam.Razor
-
-#a "..\Solutions\Endjin.AzureWeekly.Wyam\bin\Debug\netstandard2.0\Endjin.AzureWeekly.WyamModules.dll"
-
-#theme AzureWeekly
-
-// Customize your settings and add new ones here
-Settings[Keys.Host] = "AzureWeekly.info";
-
-string newscollectionFilePath = @"..\Artefacts\NewsletterCollections\168.json";
-
-Pipelines.Add(
-    "Newsletter",
-    NewsTopicModule(newscollectionFilePath),
-    Razor(),
-    WriteFiles());
\ No newline at end of file
+#n Wyam.Razor
+
+#a "..\Solutions\Endjin.AzureWeekly.Wyam\bin\Debug\netstandard2.0\Endjin.AzureWeekly.WyamModules.dll"
+
+#theme AzureWeekly
+
+// Customize your settings and add new ones here
+Settings[Keys.Host] = "AzureWeekly.info";
+
+string newscollectionFilePath = @"..\Artefacts\NewsletterCollections\168.json";
+
+Pipelines.Add(
+    "Newsletter",
+    NewsTopicModule(newscollectionFilePath),
+    Razor().WithLayout("/_Index.cshtml"),

```

Comment:
The trick here is to realize that each `IDocument` output by the `NewsTopicModule` is similar to a file, but because the documents being output don't have any content, they're like totally empty files. So running the Razor engine on it's own against files with empty content wouldn't actually render anything. That's where the `WithLayout()` method comes in - it tells the Razor engine "render these documents but use the following layout". You could have also used a `_ViewStart.cshtml` file for the same purpose, but since the documents being output from the `NewsTopicModule` don't have a `IDocument.Source` set, the Razor engine doesn't know where they're (virtually) located in the file system and so wouldn't know where to look for the `_ViewStart.cshtml`.

This is a common pattern - I use it all over the place in [Discover .NET](https://discoverdot.net). The idea is that you create `IDocument` objects without content but with lots of metadata and then use a common layout file to render each of those documents, the layout making use of the metadata. The `RenderBody()` call in the layout ends up being a no-op. Note that if you added content to each of the output `IDocument` documents in `NewsTopicModule`, that content would be interpreted as Razor code and rendered inside the `RenderBody()` call, though I usually find it easier to put all the data for my "page" in metadata in the document and then let the layout deal with rendering it from the metadata.